src/CachePlugin.php

```diff
@@ -160,6 +163,22 @@ private function getMaxAge(ResponseInterface $response)
             return (new \DateTime($header))->getTimestamp() - (new \DateTime())->getTimestamp();
         }
 
-        return $this->defaultTtl;
+        return $this->config['default_ttl'];
+    }
+
+    /**
+     * @param array $config
+     *
+     * @return array
+     */
+    protected function configure(array $config = [])
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults($this->config);
```

Comment:
I think the best practice is to have a configureOptions method which accepts an options resolver object. You can create that in the constructor, configure it with the configure options and after that you can either use it in the constructor or later with setters.